alodataset/kitti_odometry.py

```diff
@@ -0,0 +1,273 @@
+    return np.linspace(start, end, seq_size).astype(int).tolist()
+
+
+def sequence_indices(n_samples, seq_size, skip, seq_skip):
+    for start in range(0, n_samples - (seq_size - 1) * (skip + 1), seq_skip + 1):
+        yield sequence_index(start, seq_size, skip)
+
+
+class KittiOdometryDataset(BaseDataset, SplitMixin):
+    def __init__(
+        self,
+        name="kitti_odometry",
+        sequences=None,
+        grayscale=True,
+        right_frame=True,
+        sequence_size=3,
+        skip=1,
+        sequence_skip=1,
+        **kwargs,
+    ):
+        super().__init__(name=name, **kwargs)
+
+        self.grayscale = grayscale
+        self.right_frame = right_frame
+        self.sequence_size = sequence_size
+        self.skip = skip
+        self.sequence_skip = sequence_skip
+
+        if self.sample:
+            return
+
+        self.dataset_dir = os.path.join(self.dataset_dir, "dataset")
+
+        self.sequences = self._load_sequences(sequences)
+
+        self.items: Dict[Any, Any] = {"seq_params": {}}
+
+        for seq in self.sequences:
+            # Conpute sequence parameters once and store them in the items dictionary.
+            self.items["seq_params"][seq] = {}
+            seq_params = self.items["seq_params"][seq]
+
+            # Exploit data from calib.txt
+            calib = self._load_calib(os.path.join(self.dataset_dir, "sequences", seq, "calib.txt"))
+
+            # Register sequence parameters
+            seq_params["baseline"] = calib["b_gray" if self.grayscale else "b_rgb"]
+            seq_params["left_intrinsic"] = CameraIntrinsic(np.c_[calib[f"K_cam{0 if grayscale else 2}"], np.zeros(3)])
+            if right_frame:
+                seq_params["right_intrinsic"] = CameraIntrinsic(
+                    np.c_[calib[f"K_cam{1 if grayscale else 3}"], np.zeros(3)]
+                )
+            with open(os.path.join(self.dataset_dir, "sequences", seq, "times.txt"), "r") as f:
+                seq_params["times"] = [float(x) for x in f.readlines()]
+
+            if int(seq) < 11:
+                with open(os.path.join(self.dataset_dir, "poses", f"{seq}.txt"), "r") as f:
+                    seq_params["poses"] = f.readlines()
+
+            # Compute all the items.
+            sequence_size = len(seq_params["times"])
+
+            # Compute sequence indices
+            temporal_sequences = sequence_indices(sequence_size, self.sequence_size, self.skip, self.sequence_skip)
+            self.items.update(
+                {
+                    len(self.items)
+                    - 1
+                    + idx: {
+                        "sequence": seq,
+                        "temporal_sequence": temporal_seq,
+                    }
+                    for idx, temporal_seq in enumerate(temporal_sequences)
+                }
+            )
+
+    def __len__(self):
+        if self.sample:
+            return len(self.items)
+        return len(self.times)
```

Comment:
no special case needed w.r.t. `self.sample`: in both cases, return `len(self.items) `